### PR TITLE
Change the `Id` and the `DocumentId` from `int` to `long`

### LINKED ISSUE #13997: Make DocumentId and Id properties long instead of int
### Is your feature request related to a problem? Please describe.
By default, we use `long` as the default type for the database key. But, we are still using `int` for the ContentItem.

I'll submit a PR for this.

### PR DESCRIPTION
**Pull request by MikeAlhayek**
Fix #13997

This PR will introduce a very low risk of a breaking change since the [IContentManagerSession.cs](https://github.com/OrchardCMS/OrchardCore/pull/13998/files#diff-ee9f11bb81c720dc5f2761d556326777c60b27eec71ff205f0f0f016346dcb12) interface was modified from accepting `int` to accepting `long`. It is low risk because this interface should be only used by OC code. Also, if someone is using it to pass int it should be allowed which isn't going to break anything.

Okay... It turned out there is more breaking changes due to changes to interfaces.

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Html/Migrations.cs
Comment (jtkech) [inline src/OrchardCore.Modules/OrchardCore.Html/Migrations.cs:71]: var lastDocumentId = 0L;

Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Html/Migrations.cs:71]: I think what I have done is much more readable to me.

Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Html/Migrations.cs:71]: Maybe, no problem.

Just because Analyzers suggest to replace the `Type` by `var` when it can be inferred by the right part of the assignment, so when it is not the case I'm now used to first try to update the right part.

Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Html/Migrations.cs:71]: Yes I get it. But in this case I don't think the analyzer complains because the value is set to long on a different line than initializer line. 

If the analyzer shows suggestion, I would have changed it. But I did not recall seeing one

Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Html/Migrations.cs:71]: > But in this case I don't think the analyzer complains

Indeed if it is not yet inferred from the right part, what I'm now used to do.

I don't want to suggest any personal preference (maybe temporary), only follow the analyzer so no problem when it doesn't complain, but it looks like the trend is to use `var` as much as possible.

Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Html/Migrations.cs]: No worries. You seems to like `var` better. I changed it for you... anything to make @jtkech happy lol

Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Html/Migrations.cs]: Okay cool thanks ;)

On my side I just had a case where it was not possible to use `var`, when the analyzer suggests to remove an useless initialization and just use a simple declaration, we can't use `var someName;`, I thought that the type could have been inferred from the first following assignment.

Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Html/Migrations.cs]: All good. Feel free to approve and/or merge

Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Html/Migrations.cs]: Okay I will look at it this week end.

If you want to take a look at my PRs related to analyzer rules, @hishamco approved one of them but I have 8 PRs related to analyzers.

Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Html/Migrations.cs]: I am actually out on a vacation for couple of weeks. I'll look when I site behind a desk again.

**Review by hishamco — APPROVED** [review]
I'm okay with the changes, but I suggest removing anything unrelated to the actual change because JK is already has several PRs

**Comment by hishamco** [comment]
We might need to triage this

**Comment by MikeAlhayek** [comment]
I need to update more objects before we merge this. Added do not merge for now.

**Comment by jtkech** [comment]
So it seems that regardless of the current `IdentityColumnSize`, the `Id` columns of index tables are always an `int` under `Sqlite` and always a `bigint` under `Sql server`.

I'm looking at the code on YesSql side.

**Comment by MikeAlhayek** [comment]
Maybe line 20 in this file https://github.com/sebastienros/yessql/blob/main/src/YesSql.Provider.Sqlite/SqliteDialect.cs

**Comment by jtkech** [comment]
Hmm maybe, what I found is that it works if we use `col => col.PrimaryKey().NotNull()` as it is used in YesSql to create the Document table, but it doesn't work if we use `col => col.PrimaryKey().Identity()` as it is currently used for the `IndexingTask` table.

And I think it doesn't work if we use `column => column.Identity().NotNull()` as it is used on YesSql side for index tables, I will try this with the `IndexingTask` table.

**Comment by jtkech** [comment]
Yes, it works with `.PrimaryKey()` but not if we use `.Identity()` which is the case on YesSql for index tables.

**Comment by jtkech** [comment]
Okay good new, for sql server it works when we configure the `DefaultIdentityColumnSize` option in appsettings.json, and we don't need to modify the `IndexingTask` migration for the `Id` column because the option is correctly taken into account, so I will revert this change.

About `Sqlite` and `PostgreSql`, looks like using a bigint for an Identity column is not yet supported, on `YesSql` side through the dialect the same command string is used.

Sqlite

        public override string IdentityColumnString => "integer primary key autoincrement";
        public override string LegacyIdentityColumnString => "integer primary key autoincrement";

PostgreSql

        public override string IdentityColumnString => "SERIAL PRIMARY KEY";
        public override string LegacyIdentityColumnString => "SERIAL PRIMARY KEY";

While for SqlServer2 different strings are used.

        public override string IdentityColumnString => "[BIGINT] IDENTITY(1,1) primary key";
        public override string LegacyIdentityColumnString => "[INT] IDENTITY(1,1) primary key";

So nothing we can do for now ;)

**Comment by jtkech** [comment]
@MikeAlhayek 

LGTM but still mising updates, check the below files (I may be wrong for some of them).

Missing Files

**Comment by jtkech** [comment]
> I am not sure there are tied to a DocumentId

Some are tied to the `Id` primary key of the related table.

Do you want me to complete the PR when I will have time?

**Comment by MikeAlhayek** [comment]
Yes, feel free to make any necessary update and complete the PR. I just don't have access to a PC at the moment.

**Comment by jtkech** [comment]
Okay, will do if I have time before you come back ;)

**Comment by jtkech** [comment]
@MikeAlhayek 

No problem, I can work on it but we first need to sync your branch with the last main, I created a PR on your repo to update the main branch from the remote tracking branch (OC main) but I don't have the write access to be able to merge it. So if you can merge my PR on your repo.

I can also start a new PR in our repo from scratch, as you want ;)

---

Yes, we also have to search on all references of the following `ISession` methods/extensions.

- `GetAsync<T>(long[] ids, ...)` part of the `ISession` interface, we don't have any but see below.

-  `GetAsync<T>(this ISession session, int[] ids, ...)`, we have some.

    This is an extension that has been created on YesSql side to not break people, we now should use the above interface method and update the related entities and interfaces using them (see below).

- `GetAsync<T>(this ISession session, long id, ...)`, we have many.

    Here we need to update the related entities and interfaces using them. About interfaces, if for a method we only change an int param to a long param it will be transparent, if still used with an int param, it will be implicitly converted to a long.

    If we change an int collection param to a long collection, we could create an extension that still accept an int collection, as done in YesSql for `GetAsync<T>(..., int[] ids, ...)`, then maybe mark this extension as obsolete and suggest to use the updated interface method, as not done in YesSql.

    So no breaking changes and then we could just mention in the docs, but in the simplest way, the list of updated interface methods.

**Comment by MikeAlhayek** [comment]
@jtkech i am using my phone. I did not see the PR that I should merge. Do you mind sharing the link here to the PR and I'll merge it.

Thanks

**Comment by jtkech** [comment]
No problem, I saw that you merged the right one in your repo, so I could merge our last OC main branch in your PR branch, so all good, I now can work on it, maybe tomorrow, will see.

**Comment by jtkech** [comment]
For now I cleaned up the releases doc before updating it.

Because it is related to more entities, interfaces and map indexes.

**Comment by MikeAlhayek** [comment]
@jtkech can you please search the solution for `IEnumerable<int>`? I think there is at least one more place in OC.Contents/AdminControler that require to be changed.

**Comment by MikeAlhayek** [comment]
@jtkech there is an instance in MenuPartDisplayDriver that needs to also be updated. 

Also, I think we should list all the interfaces/domain models that were impacted in the "Breaking Change" section in the release notes of 1.7

**Comment by jtkech** [comment]
About `MenuPartDisplayDriver` not sure, I will check it this night.

Yes about the doc, I will list here the updated entities, interfaces and map indexes.

**Comment by jtkech** [comment]
@MikeAlhayek 

Okay I updated some missing `int to long`, I updated the release notes, and I did some little tests, a fresh setup, creating a content item, a workflow and a deployment plan, I also tried to run it on an existing database having the same entities but created before the code changes.

I let you review the doc and maybe we should do more tests, in the meantime I will approve the PR.

**Review by jtkech — APPROVED** [review]

**Comment by MikeAlhayek** [comment]
@jtkech I made minor changes to the release notes. Trying to do some testing but there seems to be an issue with the `main` branch. 

I am getting 
image

**Comment by jtkech** [comment]
The main branch merged in this PR, or the main branch itself?

I'm finishing something then I will look at it

**Comment by MikeAlhayek** [comment]
The issue is in the `main` branch and seems to be affecting this PR as well. I am guessing there is something recently merged in `main` which maybe causing this issue.

**Comment by jtkech** [comment]
Okay, will look at it soon.

**Comment by jtkech** [comment]
Just tried the last main branch, did a fresh Blog setup, for now works fone on my side?

Anything to repro?

Hmm, I will try this PR branch

**Comment by MikeAlhayek** [comment]
@jtkech strange. It sounds like a tenant specific issue "maybe a specific feature". The issue seems to happen when logging in. Its seems to only happen to few sites but not all of them. I'll try to get more info tomorrow.

**Comment by jtkech** [comment]
Okay, when I switch to different branches sometimes I need to do `git clean -xdf`.

**Comment by MikeAlhayek** [comment]
Yes and we also have `npm run cleanup`. I usually use `npm run cleanup` since it only deletes `bin` and the `lib` folders. It'll keep things like `App_Data`. But with the `npm run cleanup` some tenants were experiencing the problem. I executed `git clean -xdf` all untracked files including `App_Data` were removed, a fresh tenants had no issue. I am just wondering what caused the issue I had earlier. 

Anyhow, I tested this PR again and everything seems to be working as expected. I created new site with `Int32` as the default and then switched to the code in this PR and everything seems to be smooth. This may be a hard one to triage without trying it like you and I did. If you think it is safe to merge it. Go for it.

**Comment by jtkech** [comment]
Okay let's go, I will merge it ;)

Also related to `YesSql`, take a look at this "issue" #14036 about the table `schema`, I understood why, see my comment https://github.com/OrchardCMS/OrchardCore/issues/14036#issuecomment-1656353246, but not sure something should be done on `YesSql` side, if not we may have to adapt our tenant validation, but not so urgent I think.

**Comment by MikeAlhayek** [comment]
I saw the issue with the schema. We discussed it today in the Steering meeting and Sebastian commented with recommendation.

**Pull request merged**
